### PR TITLE
Fixes #719: Generated link for Forgot Password Endpoint

### LINKED ISSUE #719: Add Forgot Password endpoint
**I'm submitting a ...**
- [ ] bug report
- [x] feature request

**Current behaviour:**

Currently there is no forgot password endpoint for the user.

**Expected behaviour:**

Forgot password endpoint for the user.

__Yes__ I am working on this.

### PR DESCRIPTION
**Pull request by yashLadha**
Fixes #719 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `development` branch.
- [ ] I have added necessary documentation (if appropriate)

### Preview Link 

- **Replace XXX with your PR no**
- Link to live demo: http://pr-XXX-fossasia-badgeyay.surge.sh  

#### Changes proposed in this pull request:

- Created Endpoint with the token
- Necessary frontend redirection still to be made. Will be completed in #629

### CODE REVIEW

File: backend/blueprint/api/run.py [hunk 1]
```diff
@@ -2,15 +2,19 @@
...
-from api.controllers import generateBadges
-from api.controllers import homePage
-from api.controllers import errorHandlers
-from api.controllers import registerUser
-from api.controllers import loginUser
-from controllers import fileUploader
-from controllers import modifyUser
+from db import db
+from config import config
+from controllers import (
```
[the remark this answers is not in the document]
Comment (yashLadha) [reply backend/blueprint/api/run.py]: Already solved @vaibhavsingh97

File: backend/blueprint/api/controllers/resetUser.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
+import jwt
+import datetime
+
```
[the remark this answers is not in the document]
Comment (yashLadha) [reply backend/blueprint/api/controllers/resetUser.py]: Done :+1:

**Comment by djmgit** [comment]
@yashLadha Please rebase your branch.

**Comment by yashLadha** [comment]
@djmgit @ParthS007  Rebase branch please review it again @vaibhavsingh97

**Comment by djmgit** [comment]
@yashLadha  please rebase your branch again.

**Comment by yashLadha** [comment]
Done @djmgit :+1:

File: backend/blueprint/api/controllers/resetUser.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+        return jsonify(
+            Response(500).generateMessage(
+                str(e)))
+
+    if data and data['username']:
+        user = User.getUser(data['username'])
+        expire = datetime.datetime.utcnow() + datetime.timedelta(seconds=600)
+        token = jwt.encode(
+            {'id': user.username,
+            'exp': expire},
```
[the remark this answers is not in the document]
Comment (gabru-md) [reply backend/blueprint/api/controllers/resetUser.py]: do this
```python
jwt.encode({
'id' : ...
})
```
Put the curly brace along with the parenthesis.

[the remark this answers is not in the document]
Comment (gabru-md) [reply backend/blueprint/api/controllers/resetUser.py]:
```python
jwt.encode({
'id' : ...,
'exp' : ...
},
app.config['SECRET_KEY'])
```

bring the closing curly brace next in next line

[the remark this answers is not in the document]
Comment (gabru-md) [reply backend/blueprint/api/controllers/resetUser.py]: Try this maybe,
```python
jwt.encode({
'id' : ...,
'exp' : ...
},
app.config['SECRET_KEY'])
```

File: backend/blueprint/api/controllers/resetUser.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+            app.config['SECRET_KEY'])
```
Comment (gabru-md) [inline backend/blueprint/api/controllers/resetUser.py]: use 
```python
app.config.get('SECRET_KEY')
```
what say?

**Comment by yashLadha** [comment]
Done @gabru-md . @djmgit Plese review it and merge

**Pull request merged**